packages/vue/src/VueTolgee.ts

```diff
@@ -20,27 +24,50 @@ export const VueTolgee = {
       throw new Error('Tolgee instance not passed in options');
     }
 
-    const createTFunc = () => {
-      return (...props) => {
-        // @ts-ignore
-        const params = getTranslateProps(...props);
-        return tolgee.t(params);
-      };
-    };
-
-    const tFunc = ref(createTFunc());
-    tolgee.on('update', () => {
-      tFunc.value = createTFunc();
-    });
+    const isSSR = Boolean(options?.isSSR);
 
-    app.mixin({
-      computed: {
-        $t() {
-          return tFunc.value;
-        },
-      },
+    const reactiveContext = ref<TolgeeVueContext>({
+      tolgee: tolgee,
+      isInitialRender: isSSR,
     });
-    app.config.globalProperties.$tolgee = tolgee;
+
+    app.provide('tolgeeContext', reactiveContext);
+
+    if (isSSR) {
+      const getOriginalTolgeeInstance = (): TolgeeInstance => ({
+        ...reactiveContext.value.tolgee,
+        t: ((...args: Parameters<TolgeeT>) => {
+          const props = getTranslateProps(...args);
+          return tolgee.t({ ...props });
+        }) as TolgeeT,
+      });
+      const getTolgeeInstanceWithDeactivatedWrapper = (): TolgeeInstance => ({
+        ...reactiveContext.value.tolgee,
+        t: ((...args: Parameters<TolgeeT>) => {
+          const props = getTranslateProps(...args);
+          return tolgee.t({ ...props, noWrap: true });
+        }) as TolgeeT,
+      });
+
+      reactiveContext.value.tolgee = getTolgeeInstanceWithDeactivatedWrapper();
+
+      watch(
+        () => reactiveContext.value.isInitialRender,
+        (isInitialRender) => {
+          if (!isInitialRender) {
+            reactiveContext.value.tolgee = getOriginalTolgeeInstance();
+          }
+        }
+      );
+    }
+
+    app.config.globalProperties.$t = ((...args: Parameters<TolgeeT>) =>
```

Comment:
Will it react to language changes? If only $t is used?

Reply:
It should. I will double check it

Reply:
Yep, it works
You can test here: https://codesandbox.io/p/sandbox/adoring-visvesvaraya-lmr9lm